src/shared/sleep-config.h

```diff
@@ -39,6 +41,13 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(SleepConfig*, sleep_config_free);
 
 int parse_sleep_config(SleepConfig **sleep_config);
 
+static inline bool SLEEP_NEEDS_MEM_SLEEP(const SleepConfig *sc, SleepOperation operation) {
+        assert(sc);
+        assert(operation >= 0 && operation < _SLEEP_OPERATION_CONFIG_MAX);
+        return strv_contains(sc->states[operation], "mem") ||
+               strv_contains(sc->modes[operation], "suspend");
```

Comment:
"hybrid-sleep" ? Or, documentation is wrong?

Reply:
"suspend" in /sys/power/disk means to put the system into suspension after writing hibernation image, i.e. hybrid-sleep.

Reply:
OK. Could you add a short comment about that in the code?

Reply:
Ah, that's not `sc->states`, but `sc->modes`.
Still adding comment is welcome.